test/docker-e2e/e2e_pfm_test.go

```diff
@@ -0,0 +1,374 @@
+	s.chainB = s.buildChain(ctx, "chain-b", tag)
+	s.chainC = s.buildChain(ctx, "chain-c", tag)
+
+	s.hermes, err = relayer.NewHermes(ctx, s.client, t.Name(), s.network, 0, s.logger)
+	s.Require().NoError(err, "failed to create hermes")
+	err = s.hermes.Init(ctx, []tastoratypes.Chain{s.chainA, s.chainB, s.chainC})
+	s.Require().NoError(err, "failed to initialize hermes")
+
+	s.Require().NoError(s.hermes.CreateClients(ctx, s.chainA, s.chainB))
+	s.Require().NoError(s.hermes.CreateClients(ctx, s.chainB, s.chainC))
+
+	s.connAToB, s.chAToB = s.establishConnectionAndChannel(ctx, s.chainA, s.chainB)
+	s.connBToC, s.chBToC = s.establishConnectionAndChannel(ctx, s.chainB, s.chainC)
+
+	s.Require().NoError(s.hermes.Start(ctx), "failed to start hermes relayer")
+
+	s.txClientA, err = s.setupTxClient(ctx, s.chainA)
+	s.Require().NoError(err, "failed to setup tx client for chain-a")
+}
+
+// buildChain creates and starts a single celestia-app chain with one validator.
+func (s *PFMTestSuite) buildChain(ctx context.Context, chainID, tag string) *cosmos.Chain {
+	cfg := &dockerchain.Config{
+		Config: &testnode.Config{
+			Genesis: genesis.NewDefaultGenesis().
+				WithChainID(chainID).
+				WithValidators(genesis.NewDefaultValidator("validator-0")).
+				WithConsensusParams(testnode.DefaultConsensusParams()),
+			UniversalTestingConfig: testnode.UniversalTestingConfig{
+				TmConfig:     testnode.DefaultTendermintConfig(),
+				AppCreator:   testnode.DefaultAppCreator(),
+				AppConfig:    testnode.DefaultAppConfig(),
+				AppOptions:   testnode.DefaultAppOptions(),
+				SuppressLogs: true,
+			},
+		},
+		Image:           dockerchain.GetCelestiaImage(),
+		Tag:             tag,
+		DockerClient:    s.client,
+		DockerNetworkID: s.network,
+	}
+
+	chain, err := dockerchain.NewCelestiaChainBuilder(s.T(), cfg).Build(ctx)
+	s.Require().NoError(err, "failed to build chain %s", chainID)
+	s.Require().NoError(chain.Start(ctx), "failed to start chain %s", chainID)
+
+	return chain
+}
+
+// establishConnectionAndChannel creates a connection and ICS20 channel between two chains.
+func (s *PFMTestSuite) establishConnectionAndChannel(ctx context.Context, a, b *cosmos.Chain) (ibc.Connection, ibc.Channel) {
+	connection, err := s.hermes.CreateConnections(ctx, a, b)
+	s.Require().NoError(err, "failed to create IBC connection")
+
+	channelOpts := ibc.CreateChannelOptions{
+		SourcePortName: ibctransfertypes.PortID,
+		DestPortName:   ibctransfertypes.PortID,
+		Order:          ibc.OrderUnordered,
+		Version:        ibctransfertypes.Version,
+	}
+
+	channel, err := s.hermes.CreateChannel(ctx, a, connection, channelOpts)
+	s.Require().NoError(err, "failed to create IBC channel")
+	return connection, channel
+}
+
+// waitChannelOpen waits for a channel to be in OPEN state on both ends.
+func (s *PFMTestSuite) waitChannelOpen(ctx context.Context, src *cosmos.Chain, srcPort, srcChan string, dst *cosmos.Chain, dstPort, dstChan string) {
```

Comment:
is this required? I think when hermes creates the channel, it will be open when `hermes.CreateChannel` returns.